internal/command/sync/sync.go

```diff
@@ -0,0 +1,334 @@
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+// Package sync implements the `sync` CLI subcommand that updates the Library Manager content.
+package sync
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	cc "github.com/arduino/golang-concurrent-workers"
+	"github.com/arduino/libraries-repository-engine/internal/configuration"
+	"github.com/arduino/libraries-repository-engine/internal/feedback"
+	"github.com/arduino/libraries-repository-engine/internal/libraries"
+	"github.com/arduino/libraries-repository-engine/internal/libraries/db"
+	"github.com/arduino/libraries-repository-engine/internal/libraries/gitutils"
+	"github.com/arduino/libraries-repository-engine/internal/libraries/hash"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/spf13/cobra"
+)
+
+var config *configuration.Config
+
+// Run executes the command.
+func Run(command *cobra.Command, cliArguments []string) {
+	config = configuration.ReadConf(command.Flags())
+
+	setup(config)
+
+	var reposFile string
+	if len(cliArguments) > 0 {
+		reposFile = cliArguments[0]
+	} else {
+		reposFile = "./repos.txt"
+	}
+
+	if len(cliArguments) > 1 {
+		feedback.LogError(fmt.Errorf("Multiple arguments are not supported"))
+		os.Exit(1)
+	}
+
+	syncLibraries(reposFile)
+}
+
+func syncLibraries(reposFile string) {
+	if _, err := os.Stat(reposFile); os.IsNotExist(err) {
+		feedback.LogError(err)
+		os.Exit(1)
+	}
+
+	log.Println("Synchronizing libraries...")
+	repos, err := libraries.ListRepos(reposFile)
+	if feedback.LogError(err) {
+		os.Exit(1)
+	}
+
+	type jobContext struct {
+		id           int
+		repoMetadata *libraries.Repo
+	}
+
+	libraryDb := db.Init(config.LibrariesDB)
+
+	jobQueue := make(chan *jobContext)
+
+	pool := cc.New(4)
+	worker := func() {
+		log.Println("Started worker...")
+		for job := range jobQueue {
+			buffer := &bytes.Buffer{}
+			logger := log.New(buffer, "", log.LstdFlags|log.LUTC)
+			syncLibrary(logger, job.repoMetadata, libraryDb)
+
+			// Output log to file
+			if err := outputLogFile(logger, job.repoMetadata, buffer); err != nil {
+				logger.Printf("Error writing log file: %s", err.Error())
+			}
+
+			// Output log to stdout
+			fmt.Println(buffer.String())
+		}
+		log.Println("Completed worker!")
+	}
+	pool.Run(worker)
+	pool.Run(worker)
+	pool.Run(worker)
+	pool.Run(worker)
+	pool.Wait()
+
+	go func() {
+		id := 0
+		for _, repo := range repos {
+			jobQueue <- &jobContext{
+				id:           id,
+				repoMetadata: repo,
+			}
+			id++
+		}
+		close(jobQueue)
+	}()
+
+	for err := range pool.Errors {
+		feedback.LogError(err)
+	}
```

Comment:
This is ok for now since it's best not changing the logic during a refactoring but we must use a more idiomatic way of doing this. And then purge `golang-concurrent-workers` completely.